frontend/__tests__/pages/index.test.js

```diff
@@ -0,0 +1,143 @@
+import { useGetAuthors, useAddAuthor } from "../../api/authors";
+import { useGetBooks, useAddBook } from "../../api/books";
+import AddBookModal from "../../components/AddBookModal/AddBookModal";
+
+jest.mock(
+  "next/head",
+  () =>
+    ({ children }) =>
+      children
+);
+
+jest.mock("../../api/books");
+
+jest.mock("../../api/authors");
+
+jest.mock("../../components/AddBookModal/AddBookModal");
+
+describe("Index page", () => {
+  it("should render Index page with no books", () => {
+    useGetAuthors.mockReturnValue({});
+    useGetBooks.mockReturnValue({ books: [] });
+    useAddAuthor.mockReturnValue({ addAuthor: () => {} });
+    useAddBook.mockReturnValue({ addBook: () => {} });
+    render(<Index />);
+
+    expect(screen.getByText("My Library")).toBeInTheDocument();
+    expect(screen.getByText("DU Capstone")).toBeInTheDocument();
+    expect(screen.getByText("There is nothing here")).toBeInTheDocument();
+  });
+  it("should render Index page with a book", () => {
+    const authorsData = [
+      { id: "1", firstName: "J.K.", lastName: "Rowling" },
+      { id: "2", firstName: "JK", lastName: "Rowling" },
+    ];
+    const booksData = [
+      {
+        id: "1",
+        title: "Harry Potter and the Chamber of Secrets",
+        author: authorsData[0],
+        description:
+          "Harry Potter and the Chamber of Secrets is a 1998 young adult fantasy novel by J.K. Rowling, the second in the Harry Potter series. The story follows Harry’s tumultuous second year at Hogwarts School of Witchcraft and Wizardry, including an encounter with Voldemort, the wizard who killed Harry’s parents. Against this fantastic backdrop, Rowling examines such themes as death, fame, friendship, choice, and prejudice. Upon release, the novel became a worldwide bestseller and won several awards, including Children’s Book of the Year at the British Book Awards and the Nestlé Smarties Book Award; it was subsequently adapted into a 2002 film directed by Chris Columbus.",
+      },
+      {
+        id: "2",
+        title: "Harry Potter and the Prisoner of Azkaban",
+        author: authorsData[1],
+      },
+    ];
+    useGetAuthors.mockReturnValue({ authors: authorsData });
+    useGetBooks.mockReturnValue({ books: booksData });
+    useAddAuthor.mockReturnValue({ addAuthor: () => {} });
+    useAddBook.mockReturnValue({ addBook: () => {} });
+    render(<Index />);
+
+    expect(screen.getByText("My Library")).toBeInTheDocument();
+    expect(screen.getByText("DU Capstone")).toBeInTheDocument();
+    expect(screen.getByText(booksData[0].title)).toBeInTheDocument();
+    expect(
+      screen.getByText(
+        `${booksData[0].author.firstName} ${booksData[0].author.lastName}`
+      )
+    ).toBeInTheDocument();
+    expect(screen.getByText(booksData[1].title)).toBeInTheDocument();
+    expect(
+      screen.getByText(
+        `${booksData[1].author.firstName} ${booksData[1].author.lastName}`
+      )
```

Comment:
No need to put the variables here, when using expect within DU we are going to want you to hardcode the expected values

Reply:
I didn't comment on this in every place, so you'll more than likely have to go back and double check if you are doing this anywhere else